spec/features/user/user_cannot_view_admin_functions_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+feature 'Non-admin user' do
+  before(:each) do
+    @user = User.create(email: "user@example.com",
+                         password: "password2",
+                         full_name: "dave user",
+                         display_name: "user",
+                         role: 0)
+
+    visit root_path
+  end
+
+  scenario 'tries to visit all items and sees 404 error message' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.